### ISSUE TITLE
No documentation about way critical css inlining works.

### ISSUE THREAD

**Issue by va-stefanek** [angular/angular#42682]
### Description
There is lack of information about that how the critical css inlining works, what kind of styles getting inlined, on which styles we should focus to get the best results. From our research we have figure out that only styles to elements which are within index.html are inlined. Is that expected behaviour?

**Comment by petebacondarwin** [comment]
See https://angular.io/guide/workspace-config#styles-optimization-options

**Comment by va-stefanek** [comment]
@petebacondarwin it's shows how to enable that option, but it's not clear how it works and what kind of approach should we use to get benefits from that feature. As I have mentioned, it inlines only styles applied to the element which are within index.html e.g. body. Styles of the first of all component (in most cases AppComponent) are not inlined. Is that what we should expect from that feature?

**Comment by petebacondarwin** [comment]
I agree that we can and should do better. I was just adding that link as a reference to where the documentation should probably be updated.

**Comment by ehealthexperts-rk** [comment]
Please when documenting `inlineCritical` (or `optimization: true` in general) add a note, that this does not work together with `Content-Security-Policy: default-src 'self'`. The reason is that inlineCritical will render the html like
```
  <link rel="stylesheet" href="styles.02eb9920c7fcd373d29e.css" media="print" onload="this.media='all'">
  <noscript>
    <link rel="stylesheet" href="styles.02eb9920c7fcd373d29e.css">
  </noscript>
  ```
The onload will fail with given CSP, resulting in no CSS applied from the referenced stylesheet.
This was discovered in https://github.com/angular/angular-cli/issues/20864 but if the other is not resolved it might be relevant here too.
